spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/ConfigurationPropertyValueHintRecoveryStrategy.java

```diff
@@ -85,7 +86,7 @@ public void addProposals(String dsl, CheckPointedParseException exception, int d
 
 					File appFile = appResource.getFile();
 					Archive jarFileArchive = appFile.isDirectory() ? new ExplodedArchive(appFile) : new JarFileArchive(appFile);
-					classLoader = new ClassLoaderExposingJarLauncher(jarFileArchive).createClassLoader();
+					classLoader = new BootClassLoaderCreation(jarFileArchive, Thread.currentThread().getContextClassLoader()).createClassLoader();
```

Comment:
As I'm working on adding tests for this, I'll come up for a better strategy here. But long story short: for the current state of this PR, this is the only way I had the tests pass